<launch>
    <arg name="additional_model_path" default="$(find gazebo_rl_gym)/models"/>
    <env name="GAZEBO_MODEL_PATH" value="$(arg additional_model_path):$(optenv GAZEBO_MODEL_PATH)"/>    
    <!-- Gazebo world -->
    <arg name="world_file" default="$(find gazebo_rl_gym)/worlds/empty.world"/>
    <arg name="paused" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="debug" default="false"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(arg world_file)"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="headless" value="$(arg headless)"/>
        <arg name="debug" value="$(arg debug)"/>
    </include>

    <!-- Robot Spawner -->
    <arg name="config_file" default="$(find gazebo_rl_gym)/config/envs/warehouse.yaml"/>
    <arg name="controllers_file" default="$(find gazebo_rl_gym)/config/controllers.yaml"/>

    <node name="robot_spawner" pkg="gazebo_rl_gym" type="spawn_robots.py" output="screen">
        <param name="config_file" value="$(arg config_file)"/>
        <param name="controllers_file" value="$(arg controllers_file)"/>
    </node>
</launch>
